h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -207,12 +117,183 @@ class SearchBarController extends Controller {
       }
     };
 
+
     this._hiddenInput = insertHiddenInput(this.refs.searchBarForm);
 
+    let explanationList = [
+      {
+        matchOn: 'user',
+        title: 'user:',
+        explanation: 'search by username',
+      },
+      {
+        matchOn: 'tag',
+        title: 'tag:',
+        explanation: 'search for annotations with a tag',
+      },
+      {
+        matchOn: 'url',
+        title: 'url:',
+        explanation: 'see all annotations on a page',
+      },
+      {
+        matchOn: 'group',
+        title: 'group:',
+        explanation: 'show annotations created in a group you are a member of',
+      },
+    ].map((item)=>{ return Object.assign(item, { type: FACET_TYPE}); });
+
+    // tagSuggestions are made available by the scoped template data.
+    // see search.html.jinja2 for definition
+    let tagSuggestions = document.querySelector('.js-tag-suggestions');
+
+    if(tagSuggestions){
+      try{
+        tagSuggestions = JSON.parse(tagSuggestions.innerHTML.trim());
+      }catch(e){
+        console.error('Could not parse .js-tag-suggestions JSON content', e);
```

Comment:
If finding the tag succeeds but parsing the JSON fails this will lead to some confusing errors below because `.map` will be called on an `Element`. I would suggest pulling this out into a helper function which takes the selector for the suggestions tag and returns `Object|Array|null`